xbmc/interfaces/legacy/ListItem.cpp

```diff
@@ -396,6 +398,21 @@ namespace XBMCAddon
           else if (key == "dateadded")
             item->GetVideoInfoTag()->m_dateAdded.SetFromDBDateTime(value.c_str());
         }
+
+        // For backward compatibility.
+        // FIXME: Remove this behaviour. It should be possible to set only tvshowtitle without
+        // having mediatype implicitly changed.
+        if (item->GetVideoInfoTag()->m_type == MediaTypeNone)
+        {
+          if (!item->GetVideoInfoTag()->m_strShowTitle.empty() && item->GetVideoInfoTag()->m_iSeason == -1)
+          {
+            item->GetVideoInfoTag()->m_type = MediaTypeTvShow;
+          }
+          else if (!item->GetVideoInfoTag()->m_artist.empty())
+          {
+            item->GetVideoInfoTag()->m_type = MediaTypeMusicVideo;
+          }
```

Comment:
But does it do that for the python stuff? I think changing that would be an API change. See for instance above in `GUIInfoManager`. If the default is changed from MediaTypeNone to MediaTypeMovie, this logic (whatever it is), would change.